### PR TITLE
Added basic itest

### PR DESCRIPTION
**Pull request by oholiab**
Added basic integration test:

```
> make itest
./itest/itest.sh
Building itest image
Sending build context to Docker daemon  24.98MB
Step 1/10 : FROM ubuntu:xenial
 ---> 5f2bf26e3524
Step 2/10 : RUN apt-get update && apt-get -y install python python-pip virtualenv make llvm lsb-release apt-transport-https
 ---> Using cache
 ---> 81b96ca5d1fa
Step 3/10 : RUN apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 4052245BD4284CDD
 ---> Using cache
 ---> 813e65b87064
Step 4/10 : RUN echo "deb https://repo.iovisor.org/apt/$(lsb_release -cs) $(lsb_release -cs) main" > /etc/apt/sources.list.d/iovisor.list
 ---> Using cache
 ---> f63761d6b9af
Step 5/10 : RUN apt-get update && apt-get -y install bcc-tools
 ---> Using cache
 ---> 3af2ee27de8d
Step 6/10 : ADD . /work
 ---> Using cache
 ---> 37ebb086ac6e
Step 7/10 : WORKDIR /work
 ---> Using cache
 ---> 8986fcf14a19
Step 8/10 : RUN make docker-env
 ---> Using cache
 ---> 1780932c6473
Step 9/10 : ENTRYPOINT ["/work/run.sh"]
 ---> Using cache
 ---> 11ad434b8e4e
Step 10/10 : CMD -c example_config
 ---> Using cache
 ---> f49e9d792082
Successfully built f49e9d792082
Successfully tagged pidtree-itest:latest
Launching itest-container pidtree-itest_12408
85dffccb8e350265bb6dd73789bcc00a5a9ea964812ab388141e43f19f7c1c64
Creating test listener
Sleeping 20 for pidtree-bcc to start
Tailing output FIFO itest/itest_output_12408 to catch test traffic
Making test connection
Caught test traffic on 127.1.33.7:31337!
SUCCESS!
CLEANUP: Caught EXIT
CLEANUP: Killing container
pidtree-itest_12408
CLEANUP: Removing FIFO
```

This will launch a pidtree-bcc instance that's listening for loopback connections, fork off a loopback connection and then check that the log output caught it. Good for verifying that our bytecode compiles!

### CODE REVIEW

File: itest/itest.sh [hunk 1]
```diff
@@ -0,0 +1,79 @@
+#!/bin/bash -eE
```
Comment (oholiab) [inline itest/itest.sh]: Ugh my indentation is all over the place, hold off, will fix

File: itest/itest.sh [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+export FIFO_NAME=itest/itest_output_$$
```
Comment (piax93) [inline itest/itest.sh]: Maybe this should be added to the `.gitignore` to avoid accidental commits in case the cleanup fails

Comment (oholiab) [reply itest/itest.sh]: Good call, done

Comment (oholiab) [reply itest/itest.sh]: Good call, done

File: Makefile [hunk 3]
```diff
@@ -33,5 +33,8 @@ docker-interactive:
 	docker build -t pidtree-bcc .
 	docker run $(DOCKER_ARGS) --rm -it --entrypoint /bin/bash pidtree-bcc
 
+itest:
+	./itest/$@.sh
```
Comment (danielpops) [inline Makefile]: does that `$@` actually buy you anything as opposed to `./itest/itest.sh`?

**Comment by oholiab** [comment]
No not at all, I was futzing around trying to get it to do `itest_xenial`
matching for example and then I realised I didn't care that much... This is
vestigial so I'll remove it.

**Pull request merged**